scripts/fetch_smugmug_activity.py

```diff
@@ -0,0 +1,262 @@
+    # TODO: Figure out where to route errors
+
+
+def get_most_recent_comment(connection):
+    cursor = connection.cursor()
+    query = '''
+        select comment_id, created_at
+        from smugmug_comments
+        order by created_at desc
+        limit 1
+        '''
+
+    cursor.execute(query)
+    for (comment_id, created_at) in cursor:
+        return (comment_id, created_at)
+
+    return (None, datetime.min)
+
+
+def fetch_new_comments(last_comment_id, last_comment_created_at):
+    response = requests.get(COMMENT_FEED_URL)
+    xml_root = ElementTree.fromstring(response.content)
+
+    new_comments = []
+    for entry in _get_entries_from_xml_root(xml_root):
+        try:
+            comment = parse_comment_from_entry(entry)
+
+            # Stop if we have found the most recently cached comment, or if we have
+            # progressed before it in time (in case the comment we are looking for
+            # was deleted on SmugMug)
+            if (comment['comment_id'] == last_comment_id  or
+                comment['created_at'] <= last_comment_created_at):
+                break
+
+            new_comments.append(comment)
+
+        except Exception as e:
+            print(e)
+
+    # TODO: Fetch more comments if we haven't found the most recent one
+    # Comments are read in reverse chronologically, so reverse the order
+    new_comments.reverse()
+    return new_comments
+
+
+def parse_comment_from_entry(entry):
+    comment = {}
+
+    smugmug_id = _get_item_from_element(entry, 'id').text
+    matches = re.search(r'\Asmugmug:comment:(?P<comment_id>\d+)\Z', smugmug_id)
+    comment['comment_id'] = matches.groupdict()['comment_id']
+
+    title = _get_item_from_element(entry, 'title').text
+    matches = re.search(r'\A(?P<author>.*) commented on (.*) photo\Z', title)
+    comment['author'] = matches.groupdict()['author']
+
+    content = _get_item_from_element(entry, 'content').text
+    matches = re.search(
+        r'<img src="(?P<thumbnail_url>https:\/\/[^"]+\.[\w]+)".*>.*<p>Comment: (?P<comment>.*)<\/p>',
+        content)
+    matches_dict = matches.groupdict()
+    comment['thumbnail_url'] = matches_dict['thumbnail_url']
+    comment['comment'] = matches_dict['comment']
+
+    comment['comment_url'] = _get_item_from_element(entry, 'link').get('href')
+
+    timestamp = _get_item_from_element(entry, 'updated').text
+    comment['created_at'] = _get_utc_datetime_from_timestamp(timestamp)
+
+    return comment
+
+
+def insert_comments(connection, new_comments):
+    cursor = connection.cursor()
+    query = '''
+        insert into smugmug_comments
+            (comment_id, comment_url, thumbnail_url, author, comment, created_at)
+        values
+            (%(comment_id)s, %(comment_url)s, %(thumbnail_url)s, %(author)s, %(comment)s, %(created_at)s)
+        '''
+
+    # TODO: Stop SQL injection
```

Comment:
I think executemany effectively uses prepared statements, which should probably remove the main risk here?

https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursor-executemany.html